js/modules/k6/execution/execution.go

```diff
@@ -297,72 +296,67 @@ func optionsAsObject(rt *goja.Runtime, options lib.Options) (*goja.Object, error
 }
 
 type tagsDynamicObject struct {
-	Runtime *goja.Runtime
-	State   *lib.State
+	runtime *goja.Runtime
+	state   *lib.State
 }
 
 // Get a property value for the key. May return nil if the property does not exist.
 func (o *tagsDynamicObject) Get(key string) goja.Value {
-	tag, ok := o.State.Tags.GetCurrentValues().Get(key)
-	if !ok {
-		return nil
+	tcv := o.state.Tags.GetCurrentValues()
+	if tag, ok := tcv.Get(key); ok {
+		return o.runtime.ToValue(tag)
 	}
-	return o.Runtime.ToValue(tag)
+	return nil
 }
 
-// Set a property value for the key. It returns true if succeed.
-// String, Boolean and Number types are implicitly converted
-// to the goja's relative string representation.
-// In any other case, if the Throw option is set then an error is raised
-// otherwise just a Warning is written.
+// Set a property value for the key. It returns true if succeed. String, Boolean
+// and Number types are implicitly converted to the goja's relative string
+// representation. Objects are checked and accepted if they contain the index
+// and value keys. In any other case, if the Throw option is set then an error
+// is raised otherwise just a Warning is written.
```

Comment:
This comment about Throw is invalid, see https://github.com/grafana/k6/issues/2653
I think this PR copied the parts of https://github.com/grafana/k6/pull/2654 that fixed the issue? Though I see that even in my PR I forgot to adjust the comment :sweat_smile:

Reply:
Yeah, I would split the fix in one more commit, going to do soon